python/nav/web/api/v1/views.py

```diff
@@ -1153,3 +1157,105 @@ class ModuleViewSet(NAVAPIMixin, viewsets.ReadOnlyModelViewSet):
+    as a JSON array. This supports multiple MAC addresses.
+
+    Responds with a JSON dict mapping the MAC addresses to the corresponding vendors.
+    The MAC addresses will have the format `aa:bb:cc:dd:ee:ff`. If the vendor for a
+    given MAC address is not found, it will be omitted from the response.
+
+    Example GET request: `/api/1/vendor/?mac=aa:bb:cc:dd:ee:ff`
+
+    Example GET response: `{"aa:bb:cc:dd:ee:ff": "Vendor A"}`
+
+    Example POST request:
+        `/api/1/vendor/` with body `["aa:bb:cc:dd:ee:ff", "11:22:33:44:55:66"]`
+
+    Example POST response:
+        `{"aa:bb:cc:dd:ee:ff": "Vendor A", "11:22:33:44:55:66": "Vendor B"}`
+    """
+
+    @staticmethod
+    def get(request):
+        mac = request.GET.get('mac', None)
+        if not mac:
+            return Response("Missing MAC address", status=status.HTTP_400_BAD_REQUEST)
+        try:
+            results = get_vendor_names([mac])
+        except ValueError:
+            return Response("Invalid MAC address", status=status.HTTP_400_BAD_REQUEST)
+        return Response(results)
+
+    @staticmethod
+    def post(request):
+        try:
+            mac_addresses = json.loads(request.body.decode('utf-8'))
+        except json.JSONDecodeError:
+            return Response("Invalid JSON", status=status.HTTP_400_BAD_REQUEST)
+
+        if not isinstance(mac_addresses, list):
+            return Response(
+                "JSON body must represent a list of MAC addresses",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        try:
+            results = get_vendor_names(mac_addresses)
+        except ValueError:
+            return Response("Invalid MAC address", status=status.HTTP_400_BAD_REQUEST)
+
+        return Response(results)
+
+
+@django.db.transaction.atomic
+def get_vendor_names(mac_addresses: Sequence[str]) -> dict[str, str]:
+    """Get vendor names for a list of MAC addresses.
+
+    :param mac_addresses: Sequence of MAC addresses in a valid format
+        (e.g., "aa:bb:cc:dd:ee:ff").
+    :return: A dictionary mapping MAC addresses to vendor names. If the vendor for a
+        given MAC address is not found, it will be omitted from the response.
+    """
+    cursor = django.db.connection.cursor()
+    cursor.execute(
+        """
+        CREATE TEMPORARY TABLE temp_macaddrs (
+            mac macaddr PRIMARY KEY
+        )
+        """
+    )
+    # Convert list to format executemany needs
+    params = [(mac,) for mac in mac_addresses]
+    try:
+        cursor.executemany(
+            """
+            INSERT INTO temp_macaddrs (mac)
+                VALUES (%s)
+            ON CONFLICT (mac)
+                DO NOTHING
+            """,
+            params,
+        )
+    except django.db.utils.DataError as e:
+        # Invalid mac address
+        raise ValueError(str(e))
+
+    cursor.execute(
+        """
+        SELECT mac, vendor FROM temp_macaddrs INNER JOIN oui ON trunc(mac)=oui
+        """
+    )
```

Comment:
A potential different way of approaching this is to avoid any use of a temporary table, and instead materialize an in-query/in-memory table to join with:

```sql
SELECT mac, vendor
FROM (
    VALUES
        ('aa:bb:cc:dd:ee:ff'::macaddr),
        ('11:22:33:44:55:66'::macaddr),
        ('77:88:99:aa:bb:cc'::macaddr)
) AS temp_macaddrs(mac)
INNER JOIN oui ON trunc(temp_macaddrs.mac) = oui.oui;
```

But this of course implies that you must **validate** all the incoming MAC addresses in Python before attempting to build a SQL query from them (no SQL injections, please).

Reply:
Coercing every incoming mac address to `nav.macaddress.MacAddress` before using that as input to PostgreSQL shouldn't be too difficult...

Reply:
that seems reasonable, ill give it a try

Reply:
Implemented these suggestions